libs/wingc/src/parser.rs

```diff
@@ -327,10 +327,18 @@ impl<'s> Parser<'s> {
 	}
 
 	fn build_assignment_statement(&self, statement_node: &Node) -> DiagnosticResult<StmtKind> {
-		Ok(StmtKind::Assignment {
-			variable: self.build_reference(&statement_node.child_by_field_name("name").unwrap())?,
-			value: self.build_expression(&statement_node.child_by_field_name("value").unwrap())?,
-		})
+		let reference = self.build_reference(&statement_node.child_by_field_name("name").unwrap())?;
+		if let ExprKind::Reference(r) = reference.kind {
+			Ok(StmtKind::Assignment {
+				variable: r,
+				value: self.build_expression(&statement_node.child_by_field_name("value").unwrap())?,
+			})
+		} else {
+			self.add_error(
+				"Expected a reference on the left hand side of an assignment",
```

Comment:
I am a bit confused by the use of the term "reference" here. I would expect the lhs of an assignment to just be a symbol/identifier, not a reference (a "reference" in my mind is refering to something that already exists, while this is actually creating a new symbol which references the expression on the rhs).

Reply:
Note that this function is for assignment and not a declaration. Declarations have only an identifier, but here it is a reference:

```
x = 2;
a.b = 2;
a[0] = 2;
```